core/node/eth_watch/src/event_processors/mod.rs

```diff
@@ -55,6 +55,9 @@ pub(super) trait EventProcessor: 'static + fmt::Debug + Send + Sync {
     async fn process_events(
         &mut self,
         storage: &mut Connection<'_, Core>,
+        // Note, that for a chain that settles on top of L1,
+        // l1_client and sl_client point to the same network.
+        l1_client: &dyn EthClient,
         sl_client: &dyn EthClient,
```

Comment:
I have reworked this in https://github.com/matter-labs/zksync-era/pull/3078, please pass `l1_client` to `DecentralizedUpgradesEventProcessor::new` instead of `process_events`